lib/suse/connect/client.rb

```diff
@@ -25,10 +26,43 @@ def initialize(config)
       def register!
         announce_or_update
         product = @config.product || Zypper.base_product
+
+        register_product(product, install_release_package: @config.product ? true : false)
+
+        # Only install recommended packages for base products
+        if product.isbase
+          product_hash = show_product(product)
+          product_hash.recommended_extensions.each do |ext|
+            register_product(ext)
+            register_recommended!(ext)
+          end
+        end
+
+        print_title '=> Activation successful!'
+      end
+
+      # Activate, add service and install release package for all recommended
+      # extension available
+      def register_recommended!(extension)
+        extension.recommended_extensions.each do |ext|
+          register_product(ext)
+          register_recommended!(ext)
+        end
+      end
+
+      # Activate the product, add the service and install the relase package
+      def register_product(product, install_release_package = true)
+        print_title "Activating #{product.identifier} #{product.version}", "\e[34m"
         service = activate_product(product, @config.email)
+
+        print_action 'Adding zypper service'
         System.add_service(service)
-        Zypper.install_release_package(product.identifier) if @config.product
-        print_success_message product
+
+        if install_release_package
```

Comment:
So right now this is always true, but apparently there's no spec that fails here.